src/lib/features/metrics/instance/metrics.ts

```diff
@@ -150,6 +150,23 @@ export default class ClientMetricsController extends Controller {
         // Note: Custom metrics GET endpoints are now handled by the admin API
     }
 
+    private async processPromiseResults(
+        promises: Promise<void>[],
+    ): Promise<boolean> {
+        const results = await Promise.allSettled(promises);
+        const rejected = results.filter(
+            (result): result is PromiseRejectedResult =>
+                result.status === 'rejected',
+        );
+        if (rejected.length) {
+            this.logger.warn(
+                'Some promise tasks failed',
+                rejected.map((r) => r.reason?.message || r.reason),
+            );
+        }
+        return rejected.length > 0;
```

Comment:
Would it be more intuitive to reason about this method if it returned a success state instead of a negation? I.e. `return rejected.length === 0`

Reply:
I struggled to find a better way of doing this... but I agree, returning true after  `processPromiseResults`, feels like "true, I managed to process everything", while false feels like: "nope, I couldn't process all promises". Adding a commit to change this using your suggestion